deploy.sh

```diff
@@ -26,92 +26,90 @@ ${bold}DESCRIPTION${normal}
 
 ${bold}OPTIONS${normal}
     ${bold}-t | --target${normal}       Changes the directory where the script is executed. By default it will be the current directory.
-
     ${bold}-s | --structure${normal}    Changes the path where the script should look for makefiles. By default it will be the current directory of each project.
-
     ${bold}-r | --rule${normal}         Changes the makefile rule for building projects. By default it will be 'all'.
-
     ${bold}-l | --lib${normal}          Adds an external dependency to build and install.
-
     ${bold}-d | --dependency${normal}   Adds an internal dependency to build and install from the repository.
-
     ${bold}-p | --project${normal}      Adds a project to build from the repository.
+    ${bold}-ip-memoria${normal}         Sets the IP address for the Memoria module.
+    ${bold}-ip-kernel${normal}          Sets the IP address for the Kernel module.
+    ${bold}-ip-cpu${normal}             Sets the IP address for the CPU module.
 
 ${bold}COMPATIBILITY${normal}
     The repository must be in ${bold}sisoputnfrba${normal} organization and have makefiles to compile each project or dependency.
 
 ${bold}EXAMPLE${normal}
-      ${bold}./deploy.sh${normal} ${bold}-l${normal}=mumuki/cspec ${bold}-d${normal}=sockets ${bold}-p${normal}=kernel ${bold}-p${normal}=memoria ${underline}tp-2022-1c-example${nounderline}
+    ${bold}./deploy.sh${normal} ${bold}-l${normal}=mumuki/cspec ${bold}-d${normal}=sockets ${bold}-p${normal}=kernel ${bold}-p${normal}=memoria ${underline}tp-2022-1c-example${nounderline}
 
   " | less -r
-  exit
+  exit 0
 fi
 
 TARGET=""
-case $1 in
-  -t=*|--target=*)
-    case ${1#*=} in
-      /*) TARGET="${1#*=}" ;;
-      *) TARGET="$PWD/${1#*=}" ;;
-    esac
-    shift
-  ;;
-  *)
-  ;;
-esac
-
 STRUCTURE=""
-case $1 in
-  -s=*|--structure=*)
-    STRUCTURE="${1#*=}"
-    shift
-  ;;
-  *)
-  ;;
-esac
-
 RULE="all"
-case $1 in
-  -r=*|--rule=*)
-    RULE="${1#*=}"
-    shift
-  ;;
-  *)
-  ;;
-esac
-
-if [[ $# -lt 1 ]]; then
-  echo -e "\n\n${bold}No repository specified!${normal}" >&2
-  fail
-fi
+IP_MEMORIA=""
+IP_KERNEL=""
+IP_CPU=""
 
 LIBRARIES=()
 DEPENDENCIES=()
 PROJECTS=()
 
-OPTIONS=("${@:1:$#-1}")
-for i in "${OPTIONS[@]}"
-do
-    case $i in
-        -l=*|--lib=*)
-          LIBRARIES+=("${i#*=}")
-          shift
-        ;;
-        -d=*|--dependency=*)
-          DEPENDENCIES+=("${i#*=}")
-          shift
-        ;;
-        -p=*|--project=*)
-          PROJECTS+=("${i#*=}")
-          shift
-        ;;
-        *)
-          echo -e "\n\n${bold}Invalid option:${normal} ${i}" >&2
-          fail
-        ;;
-    esac
+# Process options
+while [[ "$1" =~ ^- ]]; do
+  case "$1" in
+    -t=*|--target=*)
+      case ${1#*=} in
+        /*) TARGET="${1#*=}" ;;
+        *) TARGET="$PWD/${1#*=}" ;;
+      esac
+      shift
+    ;;
+    -s=*|--structure=*)
+      STRUCTURE="${1#*=}"
+      shift
+    ;;
+    -r=*|--rule=*)
+      RULE="${1#*=}"
+      shift
+    ;;
```

Comment:
This would allow me to pass multiple `-t`, `-s`, `-r`, `-ip-*` options even though they're meant to be passed only once 😟